framework/web/Request.php

```diff
@@ -162,7 +162,39 @@ class Request extends \yii\base\Request
      * @see getBodyParams()
      */
     public $parsers = [];
+    /**
+     * @var array The configuration for trusting security related header.
+     * The array key is a regular expression for matching the hostname.
+     * The value is a list of headers to trust.
+     * Example 1: trust all headers from domains ending in '.trusted.com'
+     * Example 2: trust just the x-forwarded-for header from domains ending in '.partial.com'
+     * ```
+     * [
+     *     '/^.*\.trusted.com$/',
+     *     '/^.*\.partial.com$/' => ['X-Forwarded-For']
+     *
+     * ]
+     * ```
+     * The default value trusts all headers from all hosts (to not break BC).
+     */
+    public $trustedHostConfig = [
+        '//'
+    ];
+    /**
+     * Lists the headers that are subject to the trusted host configuration.
+     * @see https://en.wikipedia.org/wiki/List_of_HTTP_header_fields
+     * @var array
+     */
+    public $secureHeaders = [
+        'X-Forwarded-For',
+        'X-Forwarded-Host',
+        'X-Forwarded-Proto',
+        'Front-End-Https',
+    ];
 
+    public $ipHeaders = [
```

Comment:
This is not a separate set of rules, but a configuration for `getUserIP`.
So step 1 is trusting certain headers. 
Step 2 is interpreting those headers (ie in which custom header is the users' IP stored?)